elephas/ml_model.py

```diff
@@ -198,19 +200,42 @@ def extract_features_and_predict(model_yaml: str,
                                          features_col: str,
                                          model_type: ModelType,
                                          predict_classes: bool,
+                                         batch_size: int,
                                          data):
             model = model_from_yaml(model_yaml, custom_objects)
             model.set_weights(weights.value)
             predict_function = determine_predict_function(model, model_type, predict_classes)
-            return predict_function(np.stack([from_vector(x[features_col]) for x in data]))
+            if batch_size is not None and batch_size > 0:
```

Comment:
In my daily use case I need to inference a huge dataset. With the original code I have to repartition the data into many executors, i.e. I have to spin up many cloud computing nodes to do the parallel prediction otherwise it will run into OOM.

In order to enable it to make predictions on huge dataset with smaller number of nodes, I added the `inference_batch_size` to ElephasTransformer to control number of rows of vectors to be converted to 2-d np array at a time.

Reply:
For backwards compatibility (no breaking API change) we need to make `batch_size` optional arg.